### ISSUE TITLE
libtool macros include many hardcoded `/usr/bin/file` instances

### ISSUE THREAD

**Issue by nathan-at-least** [NixOS/nixpkgs#98440]
**Describe the bug**

When running `./configure` during instantiation of my project's derivation, I see error lines such as:

```
./configure: line 8395: /usr/bin/file: No such file or directory
```

Because `./configure` is bourne shell without `set -e`, the configure script continues on the incorrect branch. I haven't determined the full impact.

In hunting down the source of this hardcoded path, I find it comes from `libtool`:

```
$ grep -r '/usr/bin/file' /nix/store/*libtool*
/nix/store/lf8b1c9m3px0bmsi4zwhcj89w38vcq7a-libtool-2.4.6/bin/libtool:              if /usr/bin/file -L $add 2> /dev/null |
/nix/store/lf8b1c9m3px0bmsi4zwhcj89w38vcq7a-libtool-2.4.6/share/aclocal/libtool.m4:    case `/usr/bin/file conftest.$ac_objext` in
/nix/store/lf8b1c9m3px0bmsi4zwhcj89w38vcq7a-libtool-2.4.6/share/aclocal/libtool.m4:      case `/usr/bin/file conftest.$ac_objext` in
/nix/store/lf8b1c9m3px0bmsi4zwhcj89w38vcq7a-libtool-2.4.6/share/aclocal/libtool.m4:      case `/usr/bin/file conftest.$ac_objext` in
/nix/store/lf8b1c9m3px0bmsi4zwhcj89w38vcq7a-libtool-2.4.6/share/aclocal/libtool.m4:    case `/usr/bin/file conftest.$ac_objext` in
/nix/store/lf8b1c9m3px0bmsi4zwhcj89w38vcq7a-libtool-2.4.6/share/aclocal/libtool.m4:    case `/usr/bin/file conftest.$ac_objext` in
/nix/store/lf8b1c9m3px0bmsi4zwhcj89w38vcq7a-libtool-2.4.6/share/aclocal/libtool.m4:    case `/usr/bin/file conftest.$ac_objext` in
/nix/store/lf8b1c9m3px0bmsi4zwhcj89w38vcq7a-libtool-2.4.6/share/aclocal/libtool.m4:    case `/usr/bin/file conftest.o` in
/nix/store/lf8b1c9m3px0bmsi4zwhcj89w38vcq7a-libtool-2.4.6/share/aclocal/libtool.m4:         case `/usr/bin/file conftest.o` in
/nix/store/lf8b1c9m3px0bmsi4zwhcj89w38vcq7a-libtool-2.4.6/share/aclocal/libtool.m4:    case `/usr/bin/file conftest.o` in
/nix/store/lf8b1c9m3px0bmsi4zwhcj89w38vcq7a-libtool-2.4.6/share/aclocal/libtool.m4:  lt_cv_file_magic_cmd='/usr/bin/file -L'
/nix/store/lf8b1c9m3px0bmsi4zwhcj89w38vcq7a-libtool-2.4.6/share/aclocal/libtool.m4:      lt_cv_file_magic_cmd=/usr/bin/file
/nix/store/lf8b1c9m3px0bmsi4zwhcj89w38vcq7a-libtool-2.4.6/share/aclocal/libtool.m4:  lt_cv_file_magic_cmd=/usr/bin/file
/nix/store/lf8b1c9m3px0bmsi4zwhcj89w38vcq7a-libtool-2.4.6/share/aclocal/libtool.m4:  lt_cv_file_magic_cmd=/usr/bin/file
/nix/store/lf8b1c9m3px0bmsi4zwhcj89w38vcq7a-libtool-2.4.6/share/libtool/configure:  lt_cv_file_magic_cmd='/usr/bin/file -L'
/nix/store/lf8b1c9m3px0bmsi4zwhcj89w38vcq7a-libtool-2.4.6/share/libtool/configure:      lt_cv_file_magic_cmd=/usr/bin/file
/nix/store/lf8b1c9m3px0bmsi4zwhcj89w38vcq7a-libtool-2.4.6/share/libtool/configure:  lt_cv_file_magic_cmd=/usr/bin/file
/nix/store/lf8b1c9m3px0bmsi4zwhcj89w38vcq7a-libtool-2.4.6/share/libtool/configure:  lt_cv_file_magic_cmd=/usr/bin/file
/nix/store/lf8b1c9m3px0bmsi4zwhcj89w38vcq7a-libtool-2.4.6/share/libtool/configure:    case `/usr/bin/file conftest.$ac_objext` in
/nix/store/lf8b1c9m3px0bmsi4zwhcj89w38vcq7a-libtool-2.4.6/share/libtool/configure:      case `/usr/bin/file conftest.$ac_objext` in
/nix/store/lf8b1c9m3px0bmsi4zwhcj89w38vcq7a-libtool-2.4.6/share/libtool/configure:      case `/usr/bin/file conftest.$ac_objext` in
/nix/store/lf8b1c9m3px0bmsi4zwhcj89w38vcq7a-libtool-2.4.6/share/libtool/configure:    case `/usr/bin/file conftest.$ac_objext` in
/nix/store/lf8b1c9m3px0bmsi4zwhcj89w38vcq7a-libtool-2.4.6/share/libtool/configure:    case `/usr/bin/file conftest.$ac_objext` in
/nix/store/lf8b1c9m3px0bmsi4zwhcj89w38vcq7a-libtool-2.4.6/share/libtool/configure:    case `/usr/bin/file conftest.$ac_objext` in
/nix/store/lf8b1c9m3px0bmsi4zwhcj89w38vcq7a-libtool-2.4.6/share/libtool/configure:    case `/usr/bin/file conftest.o` in
/nix/store/lf8b1c9m3px0bmsi4zwhcj89w38vcq7a-libtool-2.4.6/share/libtool/configure:          case `/usr/bin/file conftest.o` in
/nix/store/lf8b1c9m3px0bmsi4zwhcj89w38vcq7a-libtool-2.4.6/share/libtool/configure:    case `/usr/bin/file conftest.o` in
/nix/store/lf8b1c9m3px0bmsi4zwhcj89w38vcq7a-libtool-2.4.6/share/libtool/build-aux/ltmain.sh:                if /usr/bin/file -L $add 2> /dev/null |
```

**To Reproduce**
Steps to reproduce the behavior (with `bash`):

1. `git clone 'https://github.com/nathan-at-least/zcash'`
2. `cd ./zcash`
3. `git checkout '49c15d44d14ea5c9424a5ba434388ec12e6f9ea5'` # see: https://github.com/nathan-at-least/zcash/commit/49c15d44d14ea5c9424a5ba434388ec12e6f9ea5
4. `nix-build ./zcutil/nix-builder/zcash 2>&1 | grep '/usr/bin/file'`

**Expected behavior**
There should be no output due to the grep not detecting the problem.

**Additional context**
I looked at the `libtool` derivation and it is very concise with a default generic builder. It seems like one fix would be to patch the `libtool` source to replace all hardcoded paths with the appropriate nix dependencies, so that all downstream builds in nix proceed correctly (especially autotools packages which rely on libtool).

**Notify maintainers**

**Metadata**

```
$ nix-shell -p nix-info --run "nix-info -m"
these paths will be fetched (0.00 MiB download, 0.00 MiB unpacked):
  /nix/store/q2bh089frm32ys4gx104gc2c3p96b647-nix-info
copying path '/nix/store/q2bh089frm32ys4gx104gc2c3p96b647-nix-info' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 4.15.0-112-generic, Ubuntu, 18.04.4 LTS (Bionic Beaver)`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.7`
 - channels(n): `"nixpkgs-21.03pre243353.6d4b93323e7"`
 - nixpkgs: `/home/n/.nix-defexpr/channels/nixpkgs`
```

**Maintainer information:**

I do not understand the yaml format here, but these look like the relevant derivations:

```
  libtool = libtool_2; 
                                                                           
  libtool_1_5 = callPackage ../development/tools/misc/libtool { };         
                                                                           
  libtool_2 = callPackage ../development/tools/misc/libtool/libtool2.nix { };
```

```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by nathan-at-least** [comment]
Update: In my own package's derivation which uses `stdenv.mkDerivation` I added this attribute:

```
    preConfigure = ''sed -i 's,/usr/bin/file,${file}/bin/file,g' ./configure'';
```

Additionally `file` is in `nativeBuildInputs`. This seems to fix the issue for my package. I still believe this is a relevant issue for the libtool derivation itself, since this will impact any downstream derivation that hits those code paths.

**Comment by AndersonTorres** [comment]
@nathan-at-least it looks like an upstream bug.

**Comment by Artturin** [comment]
Fixed in 97c43828fb7e016b4ee8fe434bc4d5e0b8a8b4be
